pkg/tsdb/azuremonitor/azuremonitor-datasource.go

```diff
@@ -260,25 +260,33 @@ func (e *AzureMonitorDatasource) unmarshalResponse(res *http.Response) (AzureMon
 	return data, nil
 }
 
-func (e *AzureMonitorDatasource) parseResponse(queryRes *tsdb.QueryResult, data AzureMonitorResponse, query *AzureMonitorQuery) error {
-	if len(data.Value) == 0 {
+func (e *AzureMonitorDatasource) parseResponse(queryRes *tsdb.QueryResult, amr AzureMonitorResponse, query *AzureMonitorQuery) error {
+	if len(amr.Value) == 0 {
 		return nil
 	}
 
-	for _, series := range data.Value[0].Timeseries {
-		points := []tsdb.TimePoint{}
+	for _, series := range amr.Value[0].Timeseries {
+		//points := []tsdb.TimePoint{}
 
 		metadataName := ""
 		metadataValue := ""
 		if len(series.Metadatavalues) > 0 {
 			metadataName = series.Metadatavalues[0].Name.LocalizedValue
 			metadataValue = series.Metadatavalues[0].Value
 		}
-		metricName := formatAzureMonitorLegendKey(query.Alias, query.UrlComponents["resourceName"], data.Value[0].Name.LocalizedValue, metadataName, metadataValue, data.Namespace, data.Value[0].ID)
+		metricName := formatAzureMonitorLegendKey(query.Alias, query.UrlComponents["resourceName"], amr.Value[0].Name.LocalizedValue, metadataName, metadataValue, amr.Namespace, amr.Value[0].ID)
 
-		for _, point := range series.Data {
+		frame := data.NewFrameOfFieldTypes(query.Alias, len(series.Data), data.FieldTypeTime, data.FieldTypeFloat64)
+		frame.Fields[1].Name = metricName
+		frame.Fields[1].SetConfig(&data.FieldConfig{
+			Unit: amr.Value[0].Unit,
+		})
+
```

Comment:
Need to set RefId here (at least for now). If we stop encoding the frames too early in the process, then easy to add it when marshaling the frames (like we do when converting to protobuf with the SDK).